File: ecommerce-project/02-backend/spring-boot-ecommerce/src/main/java/com/kmorath/ecommerce/service/CheckoutService.java
package com.kmorath.ecommerce.service;

import com.kmorath.ecommerce.dto.PaymentInfo;
import com.kmorath.ecommerce.dto.Purchase;
import com.kmorath.ecommerce.dto.PurchaseResponse;
import com.stripe.exception.StripeException;
import com.stripe.model.PaymentIntent;

public interface CheckoutService {

    PurchaseResponse placeOrder(Purchase purchase);

    PaymentIntent createPaymentIntent(PaymentInfo paymentInfo) throws StripeException;
}

File: ecommerce-project/03-frontend/angular-ecommerce/src/app/services/checkout.service.ts
import { environment } from 'src/environments/environment';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Purchase } from '../dto/purchase';
import { PaymentInfo } from '../dto/payment-info';


@Injectable({
  providedIn: 'root'
})
export class CheckoutService {

  private purchaseUrl = environment.shopApiUrl + '/checkout/purchase';
  private paymentIntentUrl = environment.shopApiUrl + '/checkout/payment-intent';

  constructor(private httpClient: HttpClient) { }


  placeOrder(purchase: Purchase): Observable<any> {
    return this.httpClient.post<Purchase>(this.purchaseUrl, purchase);
  }

  createPaymentIntent(paymentInfo: PaymentInfo): Observable<any> {
    return this.httpClient.post<PaymentInfo>(this.paymentIntentUrl, paymentInfo);
  }

}
